yt/geometry/coordinates/spherical_coordinates.py

```diff
@@ -277,27 +285,105 @@ def image_axis_name(self):
+        ti = self.axis_id["theta"]
+        rmin = self.ds.domain_left_edge[ri]
+        rmax = self.ds.domain_right_edge[ri]
+        thetamin = self.ds.domain_left_edge[ti]
+        thetamax = self.ds.domain_right_edge[ti]
+        corners = [
+            (rmin, thetamin),
+            (rmin, thetamax),
+            (rmax, thetamin),
+            (rmax, thetamax),
+        ]
+
+        def to_poloidal_plane(r, theta):
+            # take a r, theta position and return
+            # cylindrical R, z coordinates
+            R = r * np.sin(theta)
+            z = r * np.cos(theta)
+            return R, z
+
+        cylindrical_corner_coords = [to_poloidal_plane(*corner) for corner in corners]
+
+        thetamin = thetamin.d
+        thetamax = thetamax.d
+
+        Rmin = min(R for R, z in cylindrical_corner_coords)
+
+        if thetamin <= np.pi / 2 <= thetamax:
+            Rmax = rmax
+        else:
+            Rmax = max(R for R, z in cylindrical_corner_coords)
+
+        zmin = min(z for R, z in cylindrical_corner_coords)
+        zmax = max(z for R, z in cylindrical_corner_coords)
+
+        return Rmin, Rmax, zmin, zmax
+
+    def _get_conic_bounds(self):
+        ri = self.axis_id["r"]
+        pi = self.axis_id["phi"]
+        rmin = self.ds.domain_left_edge[ri]
+        rmax = self.ds.domain_right_edge[ri]
+        phimin = self.ds.domain_left_edge[pi]
+        phimax = self.ds.domain_right_edge[pi]
+        corners = [
+            (rmin, phimin),
+            (rmin, phimax),
+            (rmax, phimin),
+            (rmax, phimax),
+        ]
+
+        def to_conic_plane(r, phi):
+            x = r * np.cos(phi)
+            y = r * np.sin(phi)
+            return x, y
+
+        conic_corner_coords = [to_conic_plane(*corner) for corner in corners]
+
+        phimin = phimin.d
+        phimax = phimax.d
+
+        if phimin <= np.pi <= phimax:
+            xxmin = -rmax
+        else:
+            xxmin = min(xx for xx, yy in conic_corner_coords)
+
+        if phimin <= 0 <= phimax:
+            xxmax = rmax
+        else:
+            xxmin = max(xx for xx, yy in conic_corner_coords)
+
+        if phimin <= 3 * np.pi / 2 <= phimax:
+            yymin = -rmax
+        else:
+            yymin = min(yy for xx, yy in conic_corner_coords)
+
+        if phimin <= np.pi / 2 <= phimax:
+            yymax = rmax
+        else:
+            yymax = max(yy for xx, yy in conic_corner_coords)
+
+        return xxmin, xxmax, yymin, yymax
+
     def sanitize_center(self, center, axis):
         center, display_center = super().sanitize_center(center, axis)
         name = self.axis_name[axis]
         if name == "r":
             display_center = center
         elif name == "theta":
-            display_center = (
-                0.0 * display_center[0],
-                0.0 * display_center[1],
-                0.0 * display_center[2],
-            )
+            xxmin, xxmax, yymin, yymax = self._get_conic_bounds()
```

Comment:
Since the results of this function never change, couldn't we pre-compute them at initialization and make them properties?  That would work well with a baseclass-level attribute.

Reply:
do we need to have them exposed ? I feel like bare hidden attributes would do the job just as well.

Reply:
Turns out it's not that trivial to run the computation once and for all at init because at that moment, `self.ds.domain_left_edge` is a pure numpy.ndarray instead of a unyt_array, which then causes issue downstream breaking the plot. I don't think I would be able to clean this up without a much deeper refactor.

Reply:
and to be clear I'm not proposing that we do such a refactor, or at least not in this PR.

Reply:
Oh, yeesh, yes, OK.  Good point.